src/index.js

```diff
@@ -111,7 +111,7 @@ class ESLintWebpackPlugin {
             file &&
             !files.includes(file) &&
             isMatch(file, wanted) &&
-            !isMatch(file, exclude)
+            !isMatch(file, exclude, { dot: true })
```

Comment:
Does default behavior `eslint` CLI check dot files?

Reply:
I believe so: https://github.com/eslint/eslint/blob/v7.20.0/lib/cli-engine/file-enumerator.js#L56

Reply:
Instead of attempting to mimic ESLint matching behaviour ourselves, why don't we use ESLint's API? E.g. https://eslint.org/docs/developer-guide/nodejs-api#cliengineispathignored

Reply:
Because we only want to lint files that make it into the graph.

Reply:
We are not attempting mimicking eslint's logic. If that's your goal, use eslint cli.

Reply:
Oh sorry, when i opened your link it looked like you were referring to something else. The path ignored logic should be using eslint. I think there is a test that verifies ignored files are ignored.

Reply:
I honestly don't remember why there were exclusions added here. It was originally just inclusions and eslint ignore rules excluded things. Since no one person has the entirety of the intent in their own minds, I'd imagine this is just a misunderstanding.

Reply:
@jsg2021 That's not my goal, but @alexander-akait implied that the default behaviour of `eslint`'s CLI is relevant here. The method I linked to would determine whether a given path, a path already in the graph, matches exclusion criteria passed as an option to ESLint.

But I see that the `CLIEngine` is deprecated, and it doesn't look like an equivalent is possible using their new API, so perhaps this is not a good way to go. In which case, using the same `micro`/`minimatch` options as ESLint makes the most sense.

Reply:
The originating PR is a little scarce on context: https://github.com/webpack-contrib/eslint-webpack-plugin/pull/41

I hadn't realized that `.eslintignore` was respected, I had just assumed that the existence of the `exclude` option meant that it wasn't, but hadn't actually tested that assumption.

Reply:
I guess the exclude option is to save eslint spin-up time for obvious things like node_modules?

Reply:
Exactly @jsg2021, the plugin checks all files in the graph, the `exclude` option works for` node_modules` and others things, as not everyone sets up `eslintignore` with `node_modules`.

Reply:
I see no problem accepting this PR